src/memory.h

```diff
@@ -7,17 +7,64 @@ extern int BLOCK_SIZE;    // Block size in bytes (default to 2KB)
 extern int BLOCK_COUNT;   // Number of blocks (default to 1024)
 extern int PROCESS_COUNT; // Number of processes (default to 4)
 
+typedef struct
+{
+    int block_id;        // ID of the block where the data is stored
+    int valid;           // 1 if the block is valid, 0 otherwise
+    unsigned char *data; // Data stored in the block (depends on BLOCK_SIZE)
+} CacheEntry;
```

Comment:
Update `CacheEntry` to store a pointer to the memory block, since we are going to have this `MemoryBlock` struct:
```suggestion
typedef struct
{
    int valid;              // 1 if the block is valid, 0 otherwise
    MemoryBlock *mem_block; // Pointer to the cached memory block
} CacheEntry;
```